internal/server/users.go

```diff
@@ -107,3 +107,27 @@ func (a *API) UpdateUser(c *gin.Context, r *api.UpdateUserRequest) (*api.User, e
 func (a *API) DeleteUser(c *gin.Context, r *api.Resource) (*api.EmptyResponse, error) {
 	return nil, access.DeleteIdentity(c, r.ID)
 }
+
+func (a *API) DeleteUser2(c *gin.Context, r *api.Resource) (*api.EmptyResponse, error) {
+	if err := access.CanDeleteIdentity(c, r.ID); err != nil {
+		return nil, err
+	}
+
+	if err := access.DeleteAccessKeysForUser(c, r.ID); err != nil {
+		return nil, err
+	}
+
+	if err := access.RemoveUserFromAllGroups(c, r.ID); err != nil {
+		return nil, err
+	}
+
+	if err := access.DeleteCredentialForUser(c, r.ID); err != nil {
+		return nil, err
+	}
+
+	if err := access.DeleteGrantsForUser(c, r.ID); err != nil {
+		return nil, err
+	}
+
+	return nil, access.DeleteIdentity2(c, r.ID)
```

Comment:
Functions in handler files (like this one) should just transform data if needed then call the access package. The exception to this rule is `login` logic.

So all you'll need to do here is call `return nil, access.DeleteIdentity2(c, r.ID)`